docs/_includes/Types.md

```diff
@@ -461,12 +461,25 @@ parameterized by types.  These _type parameters_ are typically
 declared inside angle brackets and can stand for any type.
 
 It is sometimes necessary to restrict these type parameters so that
-they can only be instantiated by certain families of types.  As such,
-Dafny distinguishes types that support the equality operation
-in compiled contexts.  To indicate
-that a type parameter is restricted to such _equality supporting_
-types, the name of the type parameter takes the suffix
-"`(==)`".[^fn-type-mode]  For example,
+they can only be instantiated by certain families of types, that is,
+by types that have certain properties. The following subsections
+describe the restrictions Dafny supports.
+
+Dafny has some inference support that makes certain signatures less
+cluttered (described in Section [Type Inference](#sec-type-inference)).
+In some cases, this support will infer that a type-parameter
+must be restricted in a particular way, in which case Dafny
+will add the appropriate suffix, such as `(==)`, automatically.
+
+## Equality-supporting type parameters: `T(==)`
+
+Designating a type parameter with the `(==)` suffix indicates that
+the parameter may only be replaced with types that are known to
+support equality comparisons (`==` and `!=`).
+This restriction only applies during compilation; for verification
+the `(==)` suffix does not restrict the substituted type.
```

Comment:
If a type parameter is marked with `(==)`, it should always be restricted as noted in the previous sentence. But note that all types support equality in ghost contexts.

Reply:
Oh, I just tried this and it current behaves as you described! I think that behavior is not desirable. That is, it should be that "this characteristic of a type parameter is operative for both verification and compilation", even ghost contexts never inspect this type characteristic.

Reply:
Fixed per discussion